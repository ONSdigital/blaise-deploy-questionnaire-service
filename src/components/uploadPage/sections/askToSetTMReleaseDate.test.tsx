/**
 * @jest-environment jsdom
 */

import {
    navigateToSetTMReleaseDatePageAndContinue,
    selectNoTMReleaseDateAndContinue
} from "../../../features/step_definitions/helpers/functions";
import {screen} from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import '@testing-library/jest-dom/extend-expect';

const mock = new MockAdapter(axios);

import { AuthManager } from "blaise-login-react-client";
import React from "react";

jest.mock("blaise-login-react-client");
AuthManager.prototype.loggedIn = jest.fn().mockImplementation(() => {
    return Promise.resolve(true);
});

describe("Ask to set TM release date page", () => {
    const LMSQuestionnaire = "LMS2004A.bpkg"
    const OPNQuestionnaire = "OPN2004A.bpkg"

    it("should render when an LMS questionnaire is being deployed", async () => {
        await navigateToSetTMReleaseDatePageAndContinue(LMSQuestionnaire);

        expect(screen.getByText(/Would you like to set a Totalmobile release date for questionnaire/i)).toBeDefined();
        expect(screen.queryByText(/Deployment summary/i)).not.toBeInTheDocument();
    });

    it("should not render when a non-LMS questionnaire is being deployed", async () => {
        await navigateToSetTMReleaseDatePageAndContinue(OPNQuestionnaire);

        expect(screen.queryByText(/Would you like to set a Totalmobile release date for questionnaire/i)).not.toBeInTheDocument();
        expect(screen.getByText(/Deployment summary/i)).toBeDefined();
    });

    it("should continue to deploy questionnaire if no release date has been selected", async () => {
        await navigateToSetTMReleaseDatePageAndContinue(LMSQuestionnaire);
        await selectNoTMReleaseDateAndContinue();

        expect(screen.getByText(/Deployment summary/i)).toBeDefined();
    });
});
