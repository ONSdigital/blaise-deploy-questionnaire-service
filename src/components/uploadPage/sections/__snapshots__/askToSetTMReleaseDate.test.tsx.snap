// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ask to set TM release date page should match the Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid"
      >
        <div
          class="grid__col col-8@m"
        >
          <h1
            class="u-mb-l"
          >
            Would you like to set a Totalmobile release date for questionnaire 
            <em
              class="highlight"
            >
              LMS2207T
            </em>
            ?
          </h1>
          <p>
            This is used to determine the date that selected cases will be sent to Totalmobile for field allocation. If a date is not selected, then no cases will be sent to Totalmobile.
          </p>
          <fieldset
            class="fieldset"
          >
            <legend
              class="fieldset__legend"
            />
            <div
              class="radios__items"
              id="askToSetDate"
            >
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="no"
                    name="askToSetDate"
                    props="[object Object]"
                    type="radio"
                    value="no"
                  />
                  <label
                    class="radio__label "
                    for="no"
                    id="no-label"
                  >
                    No release date
                  </label>
                </span>
              </p>
              <br />
              <p
                class="radios__item"
              >
                <span
                  class="radio"
                >
                  <input
                    class="radio__input js-radio"
                    id="yes"
                    name="askToSetDate"
                    props="[object Object]"
                    type="radio"
                    value="yes"
                  />
                  <label
                    class="radio__label "
                    for="yes"
                    id="yes-label"
                  >
                    Yes, let me specify a release date
                  </label>
                  <span
                    class="radio__other radio__other--open"
                    id="other-radio-other-wrap"
                  >
                    <label
                      class="label u-fs-s--b "
                      for="set-live-date"
                      id="other-textbox-label"
                    >
                      Please specify date
                    </label>
                    <input
                      class="input input--text input-type__input input--w-auto"
                      id="set-live-date"
                      name="set release date"
                      type="date"
                      value=""
                    />
                  </span>
                </span>
              </p>
              <br />
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid"
    >
      <div
        class="grid__col col-8@m"
      >
        <h1
          class="u-mb-l"
        >
          Would you like to set a Totalmobile release date for questionnaire 
          <em
            class="highlight"
          >
            LMS2207T
          </em>
          ?
        </h1>
        <p>
          This is used to determine the date that selected cases will be sent to Totalmobile for field allocation. If a date is not selected, then no cases will be sent to Totalmobile.
        </p>
        <fieldset
          class="fieldset"
        >
          <legend
            class="fieldset__legend"
          />
          <div
            class="radios__items"
            id="askToSetDate"
          >
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="no"
                  name="askToSetDate"
                  props="[object Object]"
                  type="radio"
                  value="no"
                />
                <label
                  class="radio__label "
                  for="no"
                  id="no-label"
                >
                  No release date
                </label>
              </span>
            </p>
            <br />
            <p
              class="radios__item"
            >
              <span
                class="radio"
              >
                <input
                  class="radio__input js-radio"
                  id="yes"
                  name="askToSetDate"
                  props="[object Object]"
                  type="radio"
                  value="yes"
                />
                <label
                  class="radio__label "
                  for="yes"
                  id="yes-label"
                >
                  Yes, let me specify a release date
                </label>
                <span
                  class="radio__other radio__other--open"
                  id="other-radio-other-wrap"
                >
                  <label
                    class="label u-fs-s--b "
                    for="set-live-date"
                    id="other-textbox-label"
                  >
                    Please specify date
                  </label>
                  <input
                    class="input input--text input-type__input input--w-auto"
                    id="set-live-date"
                    name="set release date"
                    type="date"
                    value=""
                  />
                </span>
              </span>
            </p>
            <br />
          </div>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
